File: go/src/292.nim-游戏.go
/*
 * @lc app=leetcode.cn id=292 lang=golang
 *
 * [292] Nim 游戏
 */

// @lc code=start
// 如果能留给对手4个，就能赢
// 5、6、7  9、10、11 ……
// 每次拿完总能留给对手4n个
func canWinNim(n int) bool {
	// return n % 4 != 0
	return n & 3 > 0
}
// @lc code=end

File: rust/src/292.nim-游戏.rs
/*
 * @lc app=leetcode.cn id=292 lang=rust
 *
 * [292] Nim 游戏
 */

// @lc code=start
impl Solution {
    pub fn can_win_nim(n: i32) -> bool {
        n & 3 > 0
    }
}
// @lc code=end
